xbmc/music/MusicDatabase.cpp

```diff
@@ -7491,6 +7713,23 @@ void CMusicDatabase::UpdateTables(int version)
     // and filled as part of scanning anyway so simply force full rescan.
     MigrateSources();
   }
+  if (version < 73)
+  {
+    // add bBoxedSet, add strDiscSubtitles to song table
+    m_pDS->exec("ALTER TABLE album ADD bBoxedSet INTEGER \n");
+    m_pDS->exec("ALTER TABLE song ADD strDiscSubtitle TEXT \n");
+  }
+
+  if (version < 74)
+  {
+    //  add iDiscTotal to album table
+    m_pDS->exec("ALTER TABLE album ADD iDiscTotal INTEGER \n");
+    // populate iDiscTotal from the data already in the song table
+    m_pDS->exec("UPDATE album SET idisctotal = (SELECT numberofdiscs "
+                "FROM (SELECT idalbum AS id, count(DISTINCT (itrack >> 16)) "
+                "AS numberofdiscs FROM song GROUP BY idalbum) AS temp2 "
+                "WHERE album.idalbum = id) \n");
```

Comment:
Actually, mine is 4ms faster in sqlite than yours, but yours is a cleaner line I think and easier to follow for others in the future although mariadb does NOT like that query at all and I had to kill it after 30 seconds of execution, so no clue what it was doing (db has the right info in it) but there was certainly something going on. Not sure I dare let it run to conclusion to see what results it returns if it hasn't finished after 30 secs !!

Reply:
Not sure why MariaDB/MySQL should be upset, it is standard ANSI SQL. I will wind up MySQL and take a look tomorrow.

Reply:
OK, MySQL doesn't like my suggestion does it. It does run, but takes 10s compared to 0.03s on my i5 so that's a bust. But I would like to see something from MySQL explaining why their optimiser can't cope with it, and what on Earth it is doing for 10s! So I'm still doing some digging so we can stick a comment in the code about it for some future dev.

Reply:
Ha, being niggled makes me persist

```
UPDATE album SET iDisctotal = (SELECT COUNT(DISTINCT (iTrack >> 16)) 
  FROM song WHERE song.IdAlbum = album.idAlbum GROUP BY idAlbum ) 
```
acceptable in both RDBMS, sufficiently efficient and easy to follow :)